kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/AdapterGenerator.kt

```diff
@@ -133,9 +133,27 @@ internal class AdapterGenerator(
     }
 
     result.addProperty(optionsProperty)
-    for (uniqueAdapter in propertyList.distinctBy { it.delegateKey }) {
+
+    val propertyListSize = propertyList.size
+    val delegateKeyCount = LinkedHashMap<DelegateKey, Int>(propertyListSize)
```

Comment:
```kotlin
val delegateKeyCount = propertyList.groupingBy { it.key }.eachCount()
```

Reply:
ohoho, that's what i was looking for. thank you.